### PR TITLE
redirect svelte to svelte-ssr if building for ssr

### PR DESCRIPTION
**Pull request by tanhauhau**
Fixes https://github.com/sveltejs/svelte/issues/6372

### CODE REVIEW

**Comment by benmccann** [comment]
Is there something we should do to make sure we're using a version of Svelte that provides https://github.com/sveltejs/svelte/pull/6416 before executing this new line of code? I'm not sure what's best (peer dep, runtime check with warning, silent runtime check, etc.)

File: index.js [hunk 1]
```diff
@@ -50,6 +50,7 @@ module.exports = function (options = {}) {
 		 */
 		resolveId(importee, importer) {
 			if (cache_emit.has(importee)) return importee;
+			if (importee === 'svelte' && compilerOptions.generate === 'ssr') return this.resolve('svelte/ssr', importer, { skipSelf: true });
```
Comment (tanhauhau) [inline index.js:53]: need to check if the svelte version supports `svelte/ssr`

Comment (dominikg) [reply index.js:53]: maybe just test if svelte/ssr resolved and only return it if it did?

**Comment by dominikg** [comment]
fyi, here's vite-plugin-svelte's implementation https://github.com/sveltejs/vite-plugin-svelte/blob/main/packages/vite-plugin-svelte/src/index.ts#L107

**Comment by benmccann** [comment]
Closing this in favor of https://github.com/sveltejs/svelte/pull/6743

**Pull request closed without merging**